web-client/components/Button.js

```diff
@@ -1,11 +1,24 @@
-function Button(props) {
+import PropTypes from 'prop-types';
+
+const Button = ({ buttonName, onClick }) => {
```

Comment:
theres two different ways of destructuring props, can be either within the function params (which is generally used if there aren't many and its a small component), or right at the start of the function 
`const {buttonName, onClick} = props;`
destructuring is preferred in ES6 and by a lot of react devs, so you dont have to do `props.onClick` everywhere, and its a bit cleaner

Reply:
makes sense, and I assume its also easier to be explicit. To confirm my understanding, what's happening here is we're saying to take `props` and map its values such that `props.buttonName` is defined as a new variable `buttonName` in our component?

Like if we wanted to, would `props.buttonName` still work? (not saying we would, just trying to understand what's going on under the hood)

Reply:
yes that first assumption is correct. https://www.w3schools.com/react/react_es6_destructuring.asp

in this case, where we are destructuring within the params of the component, props would not available as a variable, so props.buttonName wouldn't work.

but if we were to do:
```
const Button = (props) => {
   const {buttonName, onClick} = props;
```
props.buttonName or anything within props would still be accessible within the component. If you'd rather keep it like that to keep it consistant, i can change it.

Reply:
nope, this approach works and makes sense!